### PR TITLE
*: Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by tisonkun**
### What problem does this PR solve?

Migrate to GitHub Actions

### CODE REVIEW

**Comment by tisonkun** [comment]
@purelind @zhouqiang-cl do you have any idea why CLA stalls?

**Comment by kennytm** [comment]
@tisonkun try again?

**Comment by tisonkun** [comment]
@kennytm fine. CLA passed. But I'm curious why GitHub Actions doesn't triggered. Let me see...

File: .github/workflows/suite.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Suite
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (tisonkun) [inline .github/workflows/suite.yml:8]:
```suggestion

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
```

**Comment by tisonkun** [comment]
> @tisonkun try again?

I suspect it requires an approval for first-time contributor.

**Comment by tisonkun** [comment]
@kennytm I think it is a normal review approval as shown at https://internals.tidb.io/uploads/default/optimized/1X/87d24dbeec59df7227b4bb6c235e625c683efb3b_2_1380x754.png

**Comment by kennytm** [comment]
@tisonkun no the "approve to run GitHub Actions" button looks like this:

2-fs8

**Comment by lonng** [comment]
@tisonkun I am sorry for the late reply. Do you plan to continue this work?

**Comment by tisonkun** [comment]
@lonng yep. Can you trigger the actions? It is a bit weird we cannot see it is on in this PR.

**Comment by tisonkun** [comment]
By the way since failpoint be a library, we may keep a lower go version requirement - I'm not sure with this part.

**Comment by lonng** [comment]
> By the way since failpoint be a library, we may keep a lower go version requirement - I'm not sure with this part.

Failpoint doesn't rely on the dedicated feature of the higher version Golang compiler, so I prefer to depend on the lower version.

**Review by lonng — DISMISSED** [review]

**Review by lonng — DISMISSED** [review]

**Pull request merged**